manager/orchestrator/jobs/orchestrator_controlapi_integration_test.go

```diff
@@ -0,0 +1,164 @@
+package jobs
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"context"
+	"io/ioutil"
+	"net"
+	"os"
+	"time"
+
+	"google.golang.org/grpc"
+
+	"github.com/docker/swarmkit/api"
+	"github.com/docker/swarmkit/manager/controlapi"
+	"github.com/docker/swarmkit/manager/orchestrator/testutils"
+	"github.com/docker/swarmkit/manager/state/store"
+	stateutils "github.com/docker/swarmkit/manager/state/testutils"
+)
+
+var _ = Describe("Integration between the controlapi and jobs orchestrator", func() {
+	// These tests verify that the control api and the jobs orchestrator
+	// cooperate and work correctly.
+
+	var (
+		// the object store is shared between the controlapi server and the
+		// orchestrator. makes an end-run around the manager, allowing us to
+		// avoid having to start the whole manager apparatus to test this one
+		// integration point
+		s *store.MemoryStore
+		o *Orchestrator
+
+		client api.ControlClient
+
+		// we need all of these variables captured so that we can close them
+		// out in AfterEach
+		server         *controlapi.Server
+		grpcServer     *grpc.Server
+		conn           *grpc.ClientConn
+		tempUnixSocket string
+
+		// serverDone is a channel that is closed when the goroutine running
+		// the server exits.
+		serverDone <-chan struct{}
+		// orchestratorDone is likewise closed when the orchestrator exits
+		orchestratorDone <-chan struct{}
+	)
+
+	BeforeEach(func() {
+		// By statements are just extra documentation of what's going on, no
+		// functional difference besides some test log output
+		By("setting up the controlapi client and server")
+		// a lot of this setup is taken from manager/controlapi/server_test.go,
+		// but it's unexported. in any case, re-writing it in ginkgo isn't
+		// unwarranted.
+		s = store.NewMemoryStore(&stateutils.MockProposer{})
+		server = controlapi.NewServer(s, nil, nil, nil, nil)
+
+		// we need a temporary unix socket to server on
+		temp, err := ioutil.TempFile("", "test-socket")
+		// i'm not sure why we immediately close this out, i'm not well versed
+		// in file system shenanigans, but this is what the controlapi unit
+		// tests do
```

Comment:
Isn't it simply because it needs to clear the path to create the socket there?

Reply:
that's likely the case, but I was unsure.